crates/bevy_ecs/src/query/access.rs

```diff
@@ -494,12 +548,12 @@ impl<T: SparseSetIndex> FilteredAccess<T> {
     }
 
     /// Returns a vector of elements that this and `other` cannot access at the same time.
-    pub fn get_conflicts(&self, other: &FilteredAccess<T>) -> Vec<T> {
+    pub fn get_conflicts(&self, other: &FilteredAccess<T>) -> AccessConflict {
```

Comment:
This is technically breaking, but I don't think it's worth refactoring to avoid this.

Reply:
my general opinion is that fixing unsoundness takes priority over semver. there are of course ways to avoid both in the same release but getting the soundness fix out as quickly as possible is important imo.